### PR TITLE
refactor: get preview miniapp

### PR DESCRIPTION
**Pull request by minh-rakuten**
# Description
Currently cannot load all versions of miniapp in preview mode. The cause is we made get miniapp by MiniAppInfo deprecated. This PR:

- Fix getting miniapp in preview
- Update miniapp cache in preview mode
- Change security validation of hashcheck. Using miniapp version id.

## Links
This fix is required to move MINI-235 into QA.

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/USERGUIDE.md [hunk 1]
```diff
@@ -209,7 +209,7 @@ miniAppMessageBridge.setUserInfoBridgeDispatcher(userInfoBridgeDispatcher)
 
 ### #5 Create and display a Mini App
 
-Calling `MiniApp.create` with a Mini App ID object will download the latest version of the Mini App if it has not yet been downloaded. A view will then be returned which will display the Mini App.
+Calling `MiniApp.create` with a Mini App ID or `MiniAppInfo` object will download the latest version of the Mini App if it has not yet been downloaded. A view will then be returned which will display the Mini App.
```
Comment (corycaywood) [inline miniapp/USERGUIDE.md:212]: I think this isn't true, is it? Using the Mini App ID will download the latest version, but using `MiniAppInfo` will download the specific version in the info.

Comment (minh-rakuten) [reply miniapp/USERGUIDE.md:212]: Sorry, let me edit it. I didn't read the rest of doc 😆.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniApp.kt [hunk 2]
```diff
@@ -49,6 +49,25 @@ abstract class MiniApp internal constructor() {
         miniAppNavigator: MiniAppNavigator? = null
     ): MiniAppDisplay
 
+    /**
+     * Creates a mini app.
+     * The mini app is downloaded, saved and provides a [MiniAppDisplay] when successful.
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniApp.kt:54]: I think this should mention something like this:
```suggestion
     * Creates a mini app using the mini app ID and version specified in [MiniAppInfo].
     * This should only be used in "Preview Mode".
     * The mini app is downloaded, saved and provides a [MiniAppDisplay] when successful.
```

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt [hunk 3]
```diff
@@ -98,8 +112,10 @@ internal class MiniAppDownloader(
                     val response = apiClient.downloadFile(file)
                     storage.saveFile(file, baseSavePath, response.byteStream())
                 }
-                withContext(coroutineDispatcher) {
-                    launch(Job()) { storage.removeOutdatedVersionApp(appId, versionId) }
+                if (!apiClient.isPreviewMode) {
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt:115]: Nice, so preview mode can have multiple versions downloaded, but published mode can only have one version downloaded?

Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt:115]: That's right and I also change the target of hash validation from miniapp id into version id to keep cache of multiple versions in preview mode.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt [hunk 4]
```diff
@@ -41,16 +41,28 @@ internal class RealMiniApp(
...
+                basePath,
+                miniAppInfo,
+                miniAppMessageBridge,
+                miniAppNavigator,
+                miniAppCustomPermissionCache
+            )
+        }
+    }
 
-    private suspend fun executingCreate(
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt:46]: makes sense 👍

**Review by khairul-alam-licon — COMMENTED** [review]
Nice 👍 
Thank you for taking this fix 🙏

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt [hunk 5]
```diff
@@ -57,21 +67,22 @@ internal class MiniAppDownloader(
         miniAppStatus.saveDownloadedMiniApp(miniAppInfo)
     }
 
-    private fun retrieveDownloadedVersionPath(miniAppInfo: MiniAppInfo): String? {
```
[the remark this answers is not in the document]
Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt:60]: ah my mistake. It should be private.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt [hunk 6]
```diff
@@ -57,21 +67,22 @@ internal class MiniAppDownloader(
...
+    fun retrieveDownloadedVersionPath(miniAppInfo: MiniAppInfo): String? {
         val versionPath = storage.getMiniAppVersionPath(miniAppInfo.id, miniAppInfo.version.versionId)
 
         if (miniAppStatus.isVersionDownloaded(miniAppInfo.id, miniAppInfo.version.versionId, versionPath)) {
-            return if (verifier.verify(miniAppInfo.id, File(versionPath))) {
+            return if (verifier.verify(miniAppInfo.version.versionId, File(versionPath)))
```
[the remark this answers is not in the document]
Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt:74]: Using `versionId` is the same because it is unique.

**Review by khairul-alam-licon — APPROVED** [review]
Looks good to me.

**Review by corycaywood — APPROVED** [review]
Looks good 👍

**Pull request merged**